app/master/slave.py

```diff
@@ -236,6 +238,12 @@ def _expected_session_header(self):
 
         return headers
 
+    def set_last_heartbeat_time(self):
```

Comment:
Minor (can ignore): Say `update` instead of `set`. IMO `set` usually sets the value to whatever is passed in as an argument.